### PR TITLE
Update default HNSW parameter settings

### PR DESCRIPTION
**Pull request by rschu1ze**
Use better defaults for HNSW parameters `hnsw_max_connections_per_layer`, `hnsw_candidate_list_size_for_construction` and `hnsw_candidate_list_size_for_search` than the default provided by USearch. The new settings were experimentally found by @gingerwizard and me.

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by robot-clickhouse-ci-1** [comment]
*This is an automated comment for commit 5f94239f996f896b19e3a37af77b6dd664f72615 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/70873/5f94239f996f896b19e3a37af77b6dd664f72615/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Style check</td><td>Runs a set of checks to keep the code style clean. If some of tests failed, see the related log from the report</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/70873/5f94239f996f896b19e3a37af77b6dd664f72615/style_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

File: tests/queries/0_stateless/02354_vector_search_expansion_search.sql [hunk 1]
```diff
@@ -14,7 +14,15 @@ CREATE TABLE tab(id Int32, vec Array(Float32), INDEX idx vec TYPE vector_similar
 -- Generate random values but with a fixed seed (conceptually), so that the data is deterministic.
 -- Unfortunately, no random functions in ClickHouse accepts a seed. Instead, abuse the numbers table + hash functions to provide
 -- deterministic randomness.
-INSERT INTO tab SELECT number, [sipHash64(number)/18446744073709551615, wyHash64(number)/18446744073709551615] FROM numbers(370000); -- 18446744073709551615 is the biggest UInt64
+INSERT INTO tab SELECT number, [sipHash64(number)/18446744073709551615, wyHash64(number)/18446744073709551615] FROM numbers(370000); -- 18446744073709551615 is the biggest pUInt64
```
Comment (divanik) [inline tests/queries/0_stateless/02354_vector_search_expansion_search.sql:17]: What does p stand for here?

Comment (rschu1ze) [reply tests/queries/0_stateless/02354_vector_search_expansion_search.sql:17]: It's a typo, fixed.

**Comment by rschu1ze** [comment]
@divanik Approve? Thanks.

**Review by divanik — APPROVED** [review]

**Comment by rschu1ze** [comment]
ClickHouse Stateless tests (debug, s3 storage): #60480

**Pull request merged**